ni_measurement_plugin_sdk_service/_internal/parameter/encoder.py

```diff
@@ -0,0 +1,206 @@
+# enums and default values
+from enum import Enum
+from typing import Any, Dict, Sequence
+
+from google.protobuf import descriptor_pb2, descriptor_pool, message_factory
+from google.protobuf.descriptor_pb2 import FieldDescriptorProto
+
+from ni_measurement_plugin_sdk_service._internal.parameter.decoder_strategy import (
+    get_type_default,
+)
+from ni_measurement_plugin_sdk_service._internal.parameter.metadata import (
+    ParameterMetadata,
+)
+from ni_measurement_plugin_sdk_service.measurement.info import ServiceInfo
+from tests.utilities.stubs.loopback.types_pb2 import ProtobufColor
+
+
+def serialize_parameters(
+    parameter_metadata_dict: Dict[int, ParameterMetadata],
+    parameter_values: Sequence[Any],
+    service_info: ServiceInfo,
+) -> bytes:
+    """Serialize the parameter values in same order based on the metadata_dict.
+
+    Args:
+        parameter_metadata_dict (Dict[int, ParameterMetadata]): Parameter metadata by ID.
+
+        parameter_value (Sequence[Any]): Parameter values to serialize.
+
+        Service_info (ServiceInfo): Unique service name.
+
+    Returns:
+        bytes: Serialized byte string containing parameter values.
+    """
+    pool = descriptor_pool.Default()
+    message_name = "".join(char for char in service_info.service_class if char.isalpha())
+    # Tries to find a message type in pool with message_name else it creates one
+    try:
+        message_proto = pool.FindMessageTypeByName(f"{message_name}.{message_name}")
```

Comment:
> I think it would be entirely reasonable to use the service class as the package part and use "Configurations" or "Outputs" as the message part, depending on which one you are creating. 

Please do this.

_get_service_name is stripping the dots from the package name. Can you use it as-is, or is there some problem with that approach?

Name the messages "Configurations" and "Outputs", not "Inputs" and "Outputs".